diff -urN hotplug-2004_03_11.org/etc/hotplug/hotplug.functions hotplug-2004_03_11/etc/hotplug/hotplug.functions
--- hotplug-2004_03_11.org/etc/hotplug/hotplug.functions	2004-03-12 03:13:42.662424064 +0100
+++ hotplug-2004_03_11/etc/hotplug/hotplug.functions	2004-03-12 03:13:51.189127808 +0100
@@ -163,7 +163,7 @@
 	    LOADED=true
 	fi
 
-	if [ $LOADED = false ]; then
+	if [ "$LOADED" = false ]; then
 	    mesg "missing kernel or user mode driver $MODULE "
 	fi
 	if echo "$MODULE" | grep -q "usb-storage" > /dev/null 2>&1 ; then
diff -urN hotplug-2004_03_11.org/etc/hotplug/ieee1394.agent hotplug-2004_03_11/etc/hotplug/ieee1394.agent
--- hotplug-2004_03_11.org/etc/hotplug/ieee1394.agent	2004-03-12 03:13:42.662424064 +0100
+++ hotplug-2004_03_11/etc/hotplug/ieee1394.agent	2004-03-12 03:14:35.826341920 +0100
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # IEEE1394-specific hotplug policy agent.
 #
diff -urN hotplug-2004_03_11.org/etc/hotplug/input.agent hotplug-2004_03_11/etc/hotplug/input.agent
--- hotplug-2004_03_11.org/etc/hotplug/input.agent	2004-03-12 03:13:42.669423000 +0100
+++ hotplug-2004_03_11/etc/hotplug/input.agent	2004-03-12 03:14:41.987405296 +0100
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # input-specific hotplug policy agent.
 #
diff -urN hotplug-2004_03_11.org/etc/hotplug/pci.agent hotplug-2004_03_11/etc/hotplug/pci.agent
--- hotplug-2004_03_11.org/etc/hotplug/pci.agent	2004-03-12 03:13:42.663423912 +0100
+++ hotplug-2004_03_11/etc/hotplug/pci.agent	2004-03-12 03:14:53.378673560 +0100
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # PCI-specific hotplug policy agent.
 #
diff -urN hotplug-2004_03_11.org/etc/hotplug/pci.rc hotplug-2004_03_11/etc/hotplug/pci.rc
--- hotplug-2004_03_11.org/etc/hotplug/pci.rc	2004-03-12 03:13:42.664423760 +0100
+++ hotplug-2004_03_11/etc/hotplug/pci.rc	2004-03-12 03:13:51.191127504 +0100
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # vim: syntax=sh
 #
 # pci.rc	mostly to recover lost boot-time pci hotplug events
@@ -45,7 +45,7 @@
 	if [ "$LISTER" = "" -o ! -f /proc/bus/pci/devices
 		 -o ! -x pci.agent ]; then
 	    echo $"** can't synthesize pci hotplug events"
-	    return
+	    return 1
 	fi
 
 	# these notifications will be handled by pcimodules
@@ -64,6 +64,7 @@
 case "$1" in
   start)
 	pci_boot_events
+	exit $?
         ;;
   stop)
 	# echo $"pci stop -- ignored"
diff -urN hotplug-2004_03_11.org/etc/hotplug/usb.agent hotplug-2004_03_11/etc/hotplug/usb.agent
--- hotplug-2004_03_11.org/etc/hotplug/usb.agent	2004-03-12 03:13:42.665423608 +0100
+++ hotplug-2004_03_11/etc/hotplug/usb.agent	2004-03-12 03:15:10.258107496 +0100
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # USB-specific hotplug policy agent.
 #
@@ -128,7 +128,7 @@
     fi
 
     # sysfs files may already be gone
-    if [ $ACTION = 'remove' ]; then
+    if [ "$ACTION" = 'remove' ]; then
 	exit 0
     fi
 
@@ -141,7 +141,7 @@
     # this could care about changing the default config, or warning
     # when the user hooked a fast device up so it runs slowly.
     TMP=$(cat $SYSFS/$DEVPATH/bNumConfigurations)
-    if [ $TMP -ne 1 -a "$ACTION" = add ]; then
+    if [ "$TMP" -ne 1 -a "$ACTION" = add ]; then
 	mesg Keeping default configuration with $SYSFS/$DEVPATH
     fi
 
@@ -424,7 +424,7 @@
     ;;
 
 remove)
-    if [ -x $REMOVER ]; then
+    if [ -x "$REMOVER" ]; then
 	$REMOVER
     fi
     rm -f $REMOVER
diff -urN hotplug-2004_03_11.org/etc/hotplug/usb.rc hotplug-2004_03_11/etc/hotplug/usb.rc
--- hotplug-2004_03_11.org/etc/hotplug/usb.rc	2004-03-12 03:13:42.667423304 +0100
+++ hotplug-2004_03_11/etc/hotplug/usb.rc	2004-03-12 03:13:51.195126896 +0100
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # vim: syntax=sh
 #
 # usb.rc	This brings the USB subsystem up and down safely.
@@ -37,7 +37,7 @@
 
 # In its currently-recommended configuration, XFree86 3.3.6 always opens
 # /dev/input/mice; so mousedev and input must be loaded before X11 starts.
-if [ $X11_USBMICE_HACK = true ]; then
+if [ "$X11_USBMICE_HACK" = true ]; then
     STATIC_MODULE_LIST="$MOUSE_MODULES $STATIC_MODULE_LIST"
 fi
 
@@ -207,7 +207,7 @@
     # hotplug events didn't fire during booting;
     # cope with devices that enumerated earlier
     # and may not have been fully configured.
-    if [ $SYNTHESIZE = true ]; then
+    if [ "$SYNTHESIZE" = true ]; then
     	usb_boot_events
     fi
 
diff -urN hotplug-2004_03_11.org/etc/hotplug.d/default/default.hotplug hotplug-2004_03_11/etc/hotplug.d/default/default.hotplug
--- hotplug-2004_03_11.org/etc/hotplug.d/default/default.hotplug	2004-03-12 03:13:42.671422696 +0100
+++ hotplug-2004_03_11/etc/hotplug.d/default/default.hotplug	2004-03-12 03:13:51.197126592 +0100
@@ -28,7 +28,9 @@
 
 # DEBUG=yes export DEBUG
 
-debug_mesg "arguments ($*) env (`env`)"
+if [ "$DEBUG" != "" ]; then
+	debug_mesg "arguments ($*) env (`env`)"
+fi
 
 #
 # Only one required argument:  event type type being dispatched.
