diff -urN hotplug-2004_04_01/etc/hotplug/usb.agent hotplug-2004_04_01.patch/etc/hotplug/usb.agent
--- hotplug-2004_04_01/etc/hotplug/usb.agent	2004-04-01 19:38:59.000000000 +0200
+++ hotplug-2004_04_01.patch/etc/hotplug/usb.agent	2004-08-07 23:40:30.932782136 +0200
@@ -344,7 +344,7 @@
 #
 if [ "$DEVPATH" != "" ]; then
   # probably, 2.6.x
-  REMOVER=/var/run/usb/$(readlink -f $SYSFS/$DEVPATH | sed -e 's;/;%;g')
+  REMOVER=/var/run/usb/$(readlink $SYSFS/$DEVPATH | sed -e 's;/;%;g')
 elif [ "$DEVICE" != "" ]; then
   # 2.4.x?
   REMOVER=/var/run/usb/$(echo $DEVICE | sed -e 's;/;%;g')
