diff -urN hotplug-2004_04_01.org/etc/hotplug/pci.rc hotplug-2004_04_01/etc/hotplug/pci.rc
--- hotplug-2004_04_01.org/etc/hotplug/pci.rc	2004-04-02 00:38:53.303487192 +0200
+++ hotplug-2004_04_01/etc/hotplug/pci.rc	2004-04-02 01:00:37.344242864 +0200
@@ -65,6 +65,7 @@
 case "$1" in
   start)
 	pci_boot_events
+	exit $?
         ;;
   stop)
 	# echo $"pci stop -- ignored"
diff -urN hotplug-2004_04_01.org/etc/hotplug.d/default/default.hotplug hotplug-2004_04_01/etc/hotplug.d/default/default.hotplug
--- hotplug-2004_04_01.org/etc/hotplug.d/default/default.hotplug	2004-04-02 00:38:53.324484000 +0200
+++ hotplug-2004_04_01/etc/hotplug.d/default/default.hotplug	2004-04-02 01:01:00.153775288 +0200
@@ -32,7 +32,7 @@
 
 # DEBUG=yes export DEBUG
 
-debug_mesg "arguments ($*) env (`env`)"
+[ -n "$DEBUG" ] && debug_mesg "arguments ($*) env (`env`)"
 
 #
 # Only one required argument:  event type type being dispatched.
