--- hotplug-2003_08_05/etc/hotplug/net.agent.wiget	2003-10-11 11:28:40.000000000 +0200
+++ hotplug-2003_08_05/etc/hotplug/net.agent	2003-10-11 11:28:50.000000000 +0200
@@ -43,7 +43,7 @@
 	    # RedHat and similar
 	    if [ -x /sbin/ifup ]; then
 		debug_mesg invoke ifup $INTERFACE
-		exec /sbin/ifup $INTERFACE
+		exec /sbin/ifup $INTERFACE hotplug
 	    else
 		mesg "how do I bring interfaces up on this distro?"
 	    fi
