--- hotplug-2003_05_01/etc/hotplug/hotplug.functions.wiget	2003-04-15 18:38:42.000000000 +0200
+++ hotplug-2003_05_01/etc/hotplug/hotplug.functions	2003-06-25 22:38:25.000000000 +0200
@@ -46,10 +46,9 @@
 # but some specialized ones (floppy based, etc) may not.  ("type -p"
 # is also a bash-ism, more robust than "which".)
 #
-AWK=`type -p gawk`
-if [ "$AWK" = "" ]; then
-    AWK=`type -p awk`
-fi
+# PLD/Linux: we guarantee there is /bin/awk (via RPM dependenties).
+#
+AWK="/bin/awk"
 
 
 #
--- hotplug-2003_05_01/etc/hotplug/pci.rc.wiget	2003-04-15 18:38:42.000000000 +0200
+++ hotplug-2003_05_01/etc/hotplug/pci.rc	2003-06-25 22:39:22.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # vim: syntax=sh
 #
 # pci.rc	mostly to recover lost boot-time pci hotplug events
@@ -62,7 +62,7 @@
 	pci_boot_events
         ;;
   stop)
-	echo $"pci stop -- ignored"
+	#echo $"pci stop -- ignored"
         ;;
   status)
 	echo $"PCI Status for kernel: "  `uname -srm`
--- hotplug-2003_05_01/etc/hotplug/usb.rc.wiget	2003-04-15 18:38:43.000000000 +0200
+++ hotplug-2003_05_01/etc/hotplug/usb.rc	2003-06-25 22:39:53.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # vim: syntax=sh
 #
 # usb.rc	This brings the USB subsystem up and down safely.
